File: hw/aimacana/u3/POO/java/rpg-system/src/main/java/ec/edu/espe/rpg/model/entities/Item.java
package ec.edu.espe.rpg.model.entities;

public abstract class Item {
    private String id;
    private String name;
    private double weight;
    private String description;
    private double baseValue;

    public Item(String id, String name, double weight, String description, double baseValue) {
        this.id = id;
        this.name = name;
        this.weight = weight;
        this.description = description;
        this.baseValue = baseValue;
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public double getWeight() { return weight; }
    public void setWeight(double weight) { this.weight = weight; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public double getBaseValue() { return baseValue; }
    public void setBaseValue(double baseValue) { this.baseValue = baseValue; }
}

File: hw/aimacana/u3/POO/typescript/rpg-system/src/server/models/Item.ts
export abstract class Item {
  private id: string;
  private name: string;
  private weight: number;
  private description: string;
  private baseValue: number;

  constructor(id: string, name: string, weight: number, description: string, baseValue: number) {
    this.id = id;
    this.name = name;
    this.weight = weight;
    this.description = description;
    this.baseValue = baseValue;
  }

  public getId(): string { return this.id; }
  public setId(id: string): void { this.id = id; }

  public getName(): string { return this.name; }
  public setName(name: string): void { this.name = name; }

  public getWeight(): number { return this.weight; }
  public setWeight(weight: number): void { this.weight = weight; }

  public getDescription(): string { return this.description; }
  public setDescription(description: string): void { this.description = description; }

  public getBaseValue(): number { return this.baseValue; }
  public setBaseValue(baseValue: number): void { this.baseValue = baseValue; }
}
